advent_of_code/2024/src/t/day14.clj

```diff
@@ -0,0 +1,101 @@
+(ns ^:test-refresh/focus t.day14
+  (:require [clojure.core.match :refer [match]]
+            [clojure.set :as set]
+            [clojure.string :as string]
+            [instaparse.core :refer [parser]]
+            [t.lib :as lib :refer [->long]]))
+
+(defn parse
+  [lines]
+  (->> lines
+       (map (fn [line]
+              (re-find #"p=(\d+),(\d+) v=(-?\d+),(-?\d+)" line)))
+       (map (fn [[_ & p]] (map ->long p)))
+       (mapv (fn [[x y dx dy]] [[y x] [dy dx]]))))
+
+(defn step
+  [width height]
+  (fn [robot]
+    (let [[[y x] [dy dx]] robot]
+      [[(mod (+ y dy) height)
+        (mod (+ x dx) width)]
```

Comment:
I ... completely forgot about modulo ... even when reading the conversation earlier I did not link it at all to my "teleport" function -_-
(feeling slightly stupid)